src/wiktextract/extractor/es/models.py

```diff
@@ -0,0 +1,113 @@
+from typing import Optional
+import json
+
+import logging
+
+from pydantic import BaseModel, Field, model_validator
+from pydantic.json_schema import GenerateJsonSchema
+
+from wiktextract.wxr_context import WiktextractContext
+
+
+class PydanticLogger:
+    wxr: Optional[WiktextractContext] = None
+
+    @classmethod
+    def debug(
+        cls, msg: str, trace: Optional[str] = None, sortid: str = "XYZunsorted"
+    ):
+        if cls.wxr:
+            cls.wxr.wtp.debug(msg, trace=trace, sortid=sortid)
+        else:
+            logging.debug(msg)
+
+
+class BaseModelWrap(BaseModel):
+    class Config:
+        extra = "ignore"
+        validate_assignment = True
```

Comment:
Assigning values to properties of a pydantic class will be validated for correct type.